Framework/API/inc/MantidAPI/IndexTypeProperty.h

```diff
@@ -0,0 +1,61 @@
+#ifndef MANTID_API_INDEXTYPEPROPERTY_H_
+#define MANTID_API_INDEXTYPEPROPERTY_H_
+
+#include "MantidAPI/DllConfig.h"
+#include <MantidKernel/PropertyWithValue.h>
+
+namespace Mantid {
+namespace API {
+
+enum IndexType { SpectrumNum = 1, WorkspaceIndex = 2 };
+
+/** IndexTypeProperty : TODO : Add support for DetectorIDs
```

Comment:
TODO removed and [this issue](https://github.com/mantidproject/mantid/issues/20013) has been created for the future implementation of the DetectorID->SpectrumIndex conversion